tensorflow/core/platform/posix/posix_file_system.cc

```diff
@@ -105,6 +105,9 @@ class PosixWritableFile : public WritableFile {
   }
 
   Status Close() override {
+    if (file_ == nullptr) {
+      return IOError(filename_, EBADF);
```

Comment:
is there a better error that explains that there is no file open?  e.g. FailedPreconditionError?  i understand wanting to keep it an IOError though.

Reply:
mmm dunno if TF has such thing on core c++ level, but from posix perspective this seems to be the right error to be returned, i.e. `Bad file number`